core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Authenticator.java

```diff
@@ -59,9 +63,11 @@ public OAuth2Authenticator(OAuth2Service service, OAuth2Config config)
                 return Optional.empty();
             }
             String principal = (String) claims.get().get(principalField);
-            return Optional.of(Identity.forUser(userMapping.mapUser(principal))
-                    .withPrincipal(new BasicPrincipal(principal))
-                    .build());
+            Identity.Builder identity = Identity.forUser(userMapping.mapUser(principal));
+            identity.withPrincipal(new BasicPrincipal(principal));
```

Comment:
`identity = identity.withPrincipal(new BasicPrincipal(principal));`?

Reply:
`withPrincipal` just modifies the builder and returns `this`, thus reassignment is unnecessary.